Inline macro attribute parsing changes after first document conversion

There appears to be a problem with the parsing of an inline macro's attributes. Consider an inline macro that's used like this:

```
example:alpha.bravo[format=test]
```

I expect the `InlineMacroProcessor` implementation to be called with a map containing `format` -> `test`. This is the case the first time `convert` or `convertFile` is called. When a subsequent conversion is performed the behaviour of the parsing changes and the map then contains `text` -> `format=test`.

The problem is illustrated by this test:

```java
package com.example;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

import org.asciidoctor.Asciidoctor;
import org.asciidoctor.ast.ContentNode;
import org.asciidoctor.extension.InlineMacroProcessor;
import org.junit.jupiter.api.Test;

class AttributeTests {
	
	@Test
	void inlineMacroAttributes() {
		Asciidoctor asciidoctor = Asciidoctor.Factory.create();
		asciidoctor.javaExtensionRegistry().inlineMacro(new InlineMacroProcessor("example") {
			
			@Override
			public Object process(ContentNode parent, String target, Map<String, Object> attributes) {
				return createPhraseNode(parent, "quoted", attributes.toString(), attributes, new HashMap<>());
			}
			
		});
		assertThat(convert(asciidoctor)).contains("{format=test}");
		assertThat(convert(asciidoctor)).contains("{format=test}");
	}
	
	private String convert(Asciidoctor asciidoctor) {
		return asciidoctor.convert("example:alpha.bravo[format=test]", Collections.emptyMap());
	}

}
```

It will fail when run with AsciidoctorJ 2.3. This problem appears to be a regression as the test passes with 2.0.0. I've 
[attached](https://github.com/asciidoctor/asciidoctorj/files/4765302/asciidoctorj-attribute-problem.zip) a small project containing the above test. `./gradlew build` should fail.

I tried to build your branch locally (with `./gradlew pTML`) and it failed:

```
> Task :asciidoctorj:signPom FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':asciidoctorj:signPom'.
> Cannot perform signing task ':asciidoctorj:signPom' because it has no configured signatory

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 10s
10 actionable tasks: 10 executed
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -13,6 +13,10 @@ For a detailed view of what has changed, refer to the {uri-repo}/commits/master[
 
 == Unreleased
 
+Bug Fixes::
+
+* Inline macro attribute parsing changes after first document conversion (@wilkinsona) (#926)
+
 == 2.3.0 (2020-05-02)
 
 Improvement::

--- a/asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/RubyHashMapDecorator.java
+++ b/asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/RubyHashMapDecorator.java
@@ -144,7 +144,7 @@ public class RubyHashMapDecorator implements Map<String, Object> {
                 return NodeConverter.createASTNode(rubyObject);
             }
             if (rubyObject instanceof RubySymbol) {
-                return ((RubySymbol) rubyObject).asJavaString();
+                return ":" + rubyObject.asString();
             }
             return JavaEmbedUtils.rubyToJava((IRubyObject) rubyValue);
         } else {

--- a/asciidoctorj-core/src/test/java/org/asciidoctor/extension/WhenTheInlineMacroProcessorRunsTwice.java
+++ b/asciidoctorj-core/src/test/java/org/asciidoctor/extension/WhenTheInlineMacroProcessorRunsTwice.java
@@ -1,0 +1,36 @@
+package org.asciidoctor.extension;
+
+import org.asciidoctor.Asciidoctor;
+import org.asciidoctor.ast.ContentNode;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThat;
+
+public class WhenTheInlineMacroProcessorRunsTwice {
+
+    // See https://github.com/asciidoctor/asciidoctorj/issues/926
+    @Test
+    public void inlineMacroAttributes() {
+        Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+        asciidoctor.javaExtensionRegistry().inlineMacro(new InlineMacroProcessor("example") {
+
+            @Override
+            public Object process(ContentNode parent, String target, Map<String, Object> attributes) {
+                return createPhraseNode(parent, "quoted", attributes.toString(), attributes, new HashMap<>());
+            }
+
+        });
+        assertThat(convert(asciidoctor), containsString("{format=test}"));
+        assertThat(convert(asciidoctor), containsString("{format=test}"));
+    }
+
+    private String convert(Asciidoctor asciidoctor) {
+        return asciidoctor.convert("example:alpha.bravo[format=test]", Collections.emptyMap());
+    }
+
+}

